### PR TITLE
Concurrentize pkg/registry/flowcontrol/ensurer/strategy.go

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
This PR makes the error handling in `pkg/registry/flowcontrol/ensurer/strategy.go` recognize that concurrent activity is not an error condition.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

@kubernetes/sig-api-machinery-bugs 
/cc @tkashem 
/cc @wojtek-t 
/cc @deads2k 
/cc @lavalamp

### CODE REVIEW

File: pkg/registry/flowcontrol/ensurer/strategy.go [hunk 1]
```diff
@@ -219,19 +219,27 @@ func ensureConfiguration(wrapper configurationWrapper, strategy ensureStrategy,
 	name := bootstrap.GetName()
 	configurationType := strategy.Name()
 
-	current, err := wrapper.Get(bootstrap.GetName())
-	if err != nil {
+	var current configurationObject
+	var err error
+	for {
```
Comment (wojtek-t) [inline pkg/registry/flowcontrol/ensurer/strategy.go:224]: What is the purpose of having this for loop?

We will have a second iteration only if the object was concurrently created, in which case do we really need to check update? It seems enough for me to rely on the other thing that concurrently created it to also update it (or worst case just wait for the next iteration in 10 second or so to do the update).

Comment (MikeSpreitzer) [reply pkg/registry/flowcontrol/ensurer/strategy.go:224]: In the case of a successful create, there is no immediate check.
The maintenance period is one minute.
The thought here was to keep the property (which the code already has except for a timing splinter) that if the object is created by another server then this server starts by doing a check and, if appropriate, attempting an update.

Comment (wojtek-t) [reply pkg/registry/flowcontrol/ensurer/strategy.go:224]: OK - I don't think we really need that, but that's not important enough for me to fight for it.
Let's have this merged.

**Pull request merged**